Mathlib/LinearAlgebra/LinearIndependent.lean

```diff
@@ -1416,6 +1451,11 @@ theorem linearIndependent_insert' {ι} {s : Set ι} {a : ι} {f : ι → V} (has
   erw [range_comp f ((↑) : s → ι)]
   simp
 
+theorem linearIndependent_insert_restrict {ι} {s : Set ι} {a : ι} {f : ι → V} (has : a ∉ s) :
+    (LinearIndependent K ((insert a s).restrict f)) ↔
+      LinearIndependent K (s.restrict f) ∧ f a ∉ Submodule.span K (f '' s) :=
+  linearIndependent_insert' has
```

Comment:
I don't think we want to restate every theorem for `restrict` folded and unfolded. If we do want to pursue `restrict`, we probably should replace rather than duplicate the existing lemmas.

An argument against trying to use `restrict` everywhere is that it doesn't work for things like `(LinearIndependent K fun b : ↥(insert x s) => (b : V))`